history/history.c

```diff
@@ -0,0 +1,1080 @@
+/*
+ * Copyright (C) 2022 VMware, Inc. All Rights Reserved.
+ *
+ * Licensed under the GNU Lesser General Public License v2.1 (the "License");
+ * you may not use this file except in compliance with the License. The terms
+ * of the License are located in the COPYING file of this distribution.
+ */
+
+#include <stdio.h>
+#include <time.h>
+
+#include <sqlite3.h>
+
+#include <rpm/rpmlib.h>
+#include <rpm/rpmdb.h>
+#include <rpm/rpmlog.h>
+#include <rpm/rpmps.h>
+#include <rpm/rpmts.h>
+
+#include "history.h"
+
+#define check_cond(COND) if(!(COND)) { \
+    fprintf(stderr, "check_cond failed in %s line %d\n", \
```

Comment:
use `pr_err`

Reply:
I want to keep the option open to use this code in a patched `rpm`, so I want to avoid dependencies on `tdnf`.